### PR TITLE
Robb/basic auth

### PR DESCRIPTION
**Pull request by ieatkimchi**
- adds [basic_auth](https://github.com/cultivatehq/basic_auth) package for simple http-auth on speaker route.
- requires following to be added to ENV_VARS for production.
```
     BASIC_AUTH_USERNAME=login
     BASIC_AUTH_PASSWORD=pass
```
- branch up to date with master
- package works via plug in controller, left 'realm' optional

basic_auth

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,4 +26,7 @@ npm-debug.log
 # variables.
 /config/*.secret.exs
 
+# .env
```
Comment (Kartstig) [inline .gitignore]: ?

File: test/cbus_elixir_web/controllers/speaker_controller_test.exs [hunk 2]
```diff
@@ -98,4 +121,18 @@ defmodule CbusElixirWeb.SpeakerControllerTest do
...
     {:ok, speaker: speaker}
   end
+
+  defp recycle_conn_auth(conn) do
+    conn =
+      conn
+      |> recycle()
+      |> using_basic_auth(@username, @password)
+
+    conn
```
Comment (Kartstig) [inline test/cbus_elixir_web/controllers/speaker_controller_test.exs]: Even simpler:

```elixir
defp recycle_conn_auth(conn) do
  conn
  |> recycle()
  |> using_basic_auth(@username, @password)
end
```

**Review by Kartstig — COMMENTED** [review]
Looks good. Saw a couple little things

**Comment by Kartstig** [comment]
Hey how do you generate your animations? Is it an OSX only app?

File: .gitignore (hunk 1 above)
Comment (ieatkimchi) [reply .gitignore]: basic_auth works with ENV_VARS so to run locally a .env file is needed.

**Comment by ieatkimchi** [comment]
@Kartstig I use [giphy_capture](https://giphy.com/apps/giphycapture) for the animations. I think [peek](https://github.com/phw/peek) is something similar for linux.

File: .gitignore (hunk 1 above)
Comment (Kartstig) [reply .gitignore]: Right right, but this is `#.env`

You're including `.env` on L30

Comment (ieatkimchi) [reply .gitignore]: the #.env is just a comment, I will edit it.

**Review by Kartstig — APPROVED** [review]
👍

**Pull request merged**